assets/js/app.js

```diff
@@ -0,0 +1,274 @@
+"use strict";
+
+(() => {
+    const apiBaseUrl = "https://api.coingecko.com/api/v3";
+    const maxReports = 5;
+    let reports = JSON.parse(localStorage.getItem("reports")) || [];
+    let allCoins = [];
+    let isFetching = false;
+
+    const fetchCoins = async () => {
+        if (isFetching) return;
+        isFetching = true;
+        try {
+            const response = await fetch(`${apiBaseUrl}/coins/markets?vs_currency=usd`);
+            if (!response.ok) throw new Error(`Error fetching data: ${response.status}`);
+            const coins = await response.json();
+            isFetching = false;
+            return coins;
+        } catch (error) {
+            console.error("API request failed:", error);
+            isFetching = false;
+            return [];
+        }
+    };
+
+    const renderHome = async () => {
+        const mainContent = document.getElementById("mainContent");
+        mainContent.innerHTML = `
+            <h2>Top Cryptocurrencies</h2>
+            <div id="selectedCoins" class="selected-container"></div>
+            <div id="coinContainer" class="grid-container"></div>
+        `;
+        try {
+            allCoins = await fetchCoins();
+            renderCoins(allCoins.slice(0, 50));
+            renderSelectedCoins();
+            setupSearch();
+        } catch (error) {
+            mainContent.innerHTML = `<h2>Error loading data</h2>`;
+            console.error("Error:", error);
+        }
+    };
+
+    const renderSelectedCoins = () => {
+        const selectedContainer = document.getElementById("selectedCoins");
+        if (reports.length === 0) {
+            selectedContainer.innerHTML = `<p>No coins selected</p>`;
+            return;
+        }
+        selectedContainer.innerHTML = reports.map((coinId) => {
+            const coin = allCoins.find((c) => c.id === coinId);
```

Comment:
something is broken in this logic as i was able to get myself into this:
![Screen Shot 2025-01-24 at 16 48 09](https://github.com/user-attachments/assets/8982eaf3-921e-47df-9c2b-e073c019ac1a)